ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ReplDumpTask.java

```diff
@@ -217,7 +224,12 @@ public int execute() {
       throw e;
     } catch (Exception e) {
       setException(e);
-      int errorCode = ErrorMsg.getErrorMsg(e.getMessage()).getErrorCode();
+      int errorCode;
+      if (e instanceof SnapshotException) {
+        errorCode = ErrorMsg.getErrorMsg("SNAPSHOT_ERROR").getErrorCode();
```

Comment:
why does snapshot error need to be treated specially?

Reply:
Answered here:
https://github.com/apache/hive/pull/2043#discussion_r633722120